### DISCUSSION TITLE
Rebuild Container: Unable to start container rebuild.

### DISCUSSION THREAD

**Discussion by Ferezoz** [community/community#8913] [Codespaces] [answered]
As the title says, I am not able to rebuild any of my Codespaces anymore, I get a notification that says `Rebuild Container: Unable to start container rebuild.` every time I try to rebuild. I already tried to disable extensions, remove the automatic installation of the dotfiles repo. I don't know what else to do, I tried multiple things and nothing works. The only way I found to rebuild is to delete the Codespace and create it again every time I do a change on my configuration.

It doesn't work either on Web or Desktop apps.

This is the notification I am getting:
![image](https://user-images.githubusercontent.com/12140600/146287521-0fc7e1f8-2f66-4afc-a48b-542f1cf57309.png)

This is the log I get whenever I rebuild:
![image](https://user-images.githubusercontent.com/12140600/146288739-f9ea1e7a-3dfd-48ad-99d3-be0e4df91a2c.png)

Another thing I saw which I am not sure is related, is that when I initially create the Codespace I see these logs at the end of the building process:
![image](https://user-images.githubusercontent.com/12140600/146287942-c1f57c3b-01ab-4364-8455-6891e6893f18.png)

**Comment by kiriDevs** [comment]
I am having the same problem at the moment and a support ticket with GitHub open. Can your (custom) Codespace at least restart properly (without recovery mode) after shutting down?

**Reply by Ferezoz** [reply]
Do you mean stopping the Codespace and connecting again? I can do that without a problem, it is just the rebuild that doesn't work. How is the recovery mode showing for you?

**Reply by kiriDevs** [reply]
You know it's in recovery mode when...
1. the codespace starts a minimalistic recovery image instead of your custom image
2. you connect and it pops up a window telling you that "This container is currently running in recovery mode." and asking if you want to "View Creation Log" (usually so you could see what's wrong in your configs, change it and rebuild

This only happens to me when on a custom image, not when I'm just using the default image.

The worst part about this is that codespaces also crash for me sometimes at the moment, so they are only usable in recovery mode for me right now - I do just normal editing, suddenly the codespace crashes, and even if there was an error which my configs (even though the initial, first build runs fine, so... there isn't) and I fixed it, I'd have to delete and re-create the codespace due to not being able to rebuild - so my changes would be gone again.

**Comment by metaskills** [comment]
> The only way I found to rebuild is to delete the Codespace and create it again every time I do a change on my configuration.

HELP! This is really frustrating. This is happening for our entire organization and every user & codespaces. Rebuilding is completely broken for us.

**Comment by tanmayeekamath** [comment]
Hey everyone, we're actively investigating this issue on our end. We will post an update here as soon as a fix is pushed. 
Apologies for the inconvenience and thanks for your patience. 

-Tanmayee

**Reply by samsarkleio** [reply]
For what it's worth we're also seeing this on Codespaces:
```
[Client:extension:1bff:resolver:resolve:1:backround-tasks I] asking user if they want to rebuild> object #0: "Rebuild Now"
[Client:extension:1bff:resolver:resolve:1:backround-tasks I] user decided to rebuild now
[Command:Rebuild I] Decorator starting: ErrorNotification
[Command:Rebuild I] Decorator starting: Telemetry
[Client:extension:1bff:liveshare:connection:8A3BE892B830C96AB678674425FA3EAE051D:liveshare:WorkspaceClient V] < IEnvironmentConfigurationService.rebuildContainer()
[Client:extension:1bff:liveshare:connection:8A3BE892B830C96AB678674425FA3EAE051D:liveshare:WorkspaceClient E] > IEnvironmentConfigurationService.rebuildContainer() error: undefined
[Command:Rebuild E] Command [Rebuild Container]: Rebuild Container Failure: Unable to start container rebuild. (1154ms)
[Command:Rebuild I] Decorator finishing: Telemetry. Duration: 7ms (0 + 7)
```

**Comment by legomushroom** [comment, marked as answer]
This is fixed by rolling back recent services deployment with the regression. Please try again, you might need to suspend/resume your existing codespace to pick up the fix, new codespaces should work as intended by default 😊 

Sorry for the trouble and thanks for reporting!

**Reply by Ferezoz** [reply]
Thank you @legomushroom, I restarted and updated VS Code, and rebuild is working now. The error at the end of the building process is still there though but it should be a different issue.

**Reply by legomushroom** [reply]
Thanks for confirming! cc @edgonmsft @eljog for the error after the rebuild
